airbyte-workers/src/main/java/io/airbyte/workers/WorkerApp.java

```diff
@@ -381,175 +381,302 @@ static Optional<ContainerOrchestratorConfig> getContainerOrchestratorConfig(fina
-    final StreamResetPersistence streamResetPersistence = new StreamResetPersistence(configDatabase);
-
-    final TemporalClient temporalClient = new TemporalClient(workflowClient, configs.getWorkspaceRoot(), temporalService, streamResetPersistence);
-    TemporalUtils.configureTemporalNamespace(temporalService);
+  private static void initializeCommonDependencies() {
+    LOGGER.debug("Initializing common worker dependencies.");
+    configs = new EnvConfigs();
+    LOGGER.info("workspaceRoot = " + configs.getWorkspaceRoot());
 
-    final TemporalWorkerRunFactory temporalWorkerRunFactory = new TemporalWorkerRunFactory(
-        temporalClient,
-        workspaceRoot,
-        configs.getAirbyteVersionOrWarning(),
-        featureFlags);
+    MetricClientFactory.initialize(MetricEmittingApps.WORKER);
 
-    final WorkspaceHelper workspaceHelper = new WorkspaceHelper(
-        configRepository,
-        jobPersistence);
+    LogClientSingleton.getInstance().setWorkspaceMdc(
+        configs.getWorkerEnvironment(),
+        configs.getLogConfigs(),
+        LogClientSingleton.getInstance().getSchedulerLogsRoot(configs.getWorkspaceRoot()));
 
-    final ConnectionHelper connectionHelper = new ConnectionHelper(configRepository, workspaceHelper);
+    if (configs.getWorkerEnvironment().equals(WorkerEnvironment.KUBERNETES)) {
+      KubePortManagerSingleton.init(configs.getTemporalWorkerPorts());
+    }
 
-    final Optional<ContainerOrchestratorConfig> containerOrchestratorConfig = getContainerOrchestratorConfig(configs);
+    featureFlags = new EnvVariableFeatureFlags();
+    defaultWorkerConfigs = new WorkerConfigs(configs);
+    temporalService = TemporalUtils.createTemporalService();
+    workflowClient = TemporalUtils.createWorkflowClient(temporalService, TemporalUtils.getNamespace());
+    TemporalUtils.configureTemporalNamespace(temporalService);
+    airbyteApiClient = getApiClient(configs);
+  }
 
-    final WebUrlHelper webUrlHelper = new WebUrlHelper(configs.getWebappUrl());
+  private static void initializeControlPlaneDependencies() throws IOException, DatabaseCheckException {
+    if (!configs.isControlPlaneWorker()) {
+      LOGGER.debug("Skipping Control Plane dependency initialization.");
+      return;
+    }
+    LOGGER.debug("Initializing control plane worker dependencies.");
+
+    final DataSource configsDataSource = DataSourceFactory.create(configs.getConfigDatabaseUser(), configs.getConfigDatabasePassword(),
+        DRIVER_CLASS_NAME, configs.getConfigDatabaseUrl());
+    final DataSource jobsDataSource = DataSourceFactory.create(configs.getDatabaseUser(), configs.getDatabasePassword(),
+        DRIVER_CLASS_NAME, configs.getDatabaseUrl());
+
+    // Manual configuration that will be replaced by Dependency Injection in the future
+    try (final DSLContext configsDslContext = DSLContextFactory.create(configsDataSource, SQLDialect.POSTGRES);
```

Comment:
I don't think I understand why there's a migration happening in this class. Can you please elaborate?

Reply:
I don't really have context on most of this, it's existing WorkerApp code that I just moved around. I _think_ the Flyway stuff here is just to perform a check that the configs database is up to date with whatever version of Airbyte the worker is running. Does that answer your question?

Reply:
Yeah, this is just how our database code is initialised today. It looks strange since it's refactored to be micronaut friendly. No migrations are run here.

Reply:
Continued in child PR here: https://github.com/airbytehq/airbyte/pull/15904/files#r953192912